lib/node_modules/@stdlib/math/base/ops/imuldw/lib/main.js

```diff
@@ -20,79 +20,24 @@
 
 // MODULES //
 
-var isnan = require( '@stdlib/math/base/assert/is-nan' );
-
-
-// VARIABLES //
-
-// Define a mask for the least significant 16 bits (low word): 65535 => 0x0000ffff => 00000000000000001111111111111111
-var LOW_WORD_MASK = 0x0000ffff>>>0; // asm type annotation
+var fcn = require( './assign.js' );
 
 
 // MAIN //
 
 /**
 * Performs multiplication of two signed 32-bit integers and returns an array of two signed 32-bit integers which represents the signed 64-bit integer product.
 *
-* @param {ArrayLikeObject} [out] - output array
 * @param {uinteger32} a - integer
 * @param {uinteger32} b - integer
-* @returns {ArrayLikeObject} output array
+* @returns {Array<number>} output array
 *
 * @example
 * var v = imuldw( 0xAAAAAAAA, 0x55555555 );
 * // returns [ -477218589, 1908874354 ]
 */
-function imuldw() {
-	var out;
-	var w1;
-	var w2;
-	var w3;
-	var ha;
-	var hb;
-	var la;
-	var lb;
-	var a;
-	var b;
-	var t;
-	var k;
-
-	if ( arguments.length === 3 ) {
-		out = arguments[ 0 ];
-		a = arguments[ 1 ];
-		b = arguments[ 2 ];
-	} else {
-		out = [ 0, 0 ];
-		a = arguments[ 0 ];
-		b = arguments[ 1 ];
-	}
-	if ( isnan( a ) || isnan( b ) ) {
-		return NaN;
-	}
-	a |= 0; // asm type annotation
-	b |= 0; // asm type annotation
-
-	ha = ( a >> 16 ) | 0;
-	la = ( a & LOW_WORD_MASK ) >>> 0;
-
-	hb = ( b >> 16 ) | 0;
-	lb = ( b & LOW_WORD_MASK ) >>> 0;
-
-	t = ( la*lb ) >>> 0;
-	w3 = ( t & LOW_WORD_MASK ) >>> 0;
-	k = ( t >>> 16 ) >>> 0;
-
-	t = ( ( ha*lb ) + k ) >>> 0;
-	w2 = ( t & LOW_WORD_MASK ) >>> 0;
-	w1 = ( t >> 16 ) >>> 0;
-
-	t = ( ( la*hb ) + w2 ) >>> 0;
-	k = ( t >> 16 ) >>> 0;
-
-	out[ 0 ] = ( ( ha*hb ) + w1 + k ) | 0; // compute the higher 32 bits and cast to a signed 32-bit integer
-	out[ 1 ] = ( ( t << 16 ) + w3 ) | 0; // compute the lower 32 bits and cast to a signed 32-bit integer
-
-	return out;
+function imuldw( a, b ) {
+	return fcn( a, b, [ 0.0, 0.0 ], 1, 0 );
```

Comment:
```suggestion
	return fcn( a, b, [ 0, 0 ], 1, 0 );
```

We should, by default, expect integer values, so we can elide the decimals here. We use decimals when we know there's a possibility of rational numbers. If an API should only return integral values, we don't use decimals.